roles/reproducer/tasks/devscripts_layout.yml

```diff
@@ -1,6 +1,224 @@
 ---
-- name: Run devscripts role
-  vars:
-    cifmw_use_reproducer: true
+- name: Ensure we have up-to-date devscripts config
+  ansible.builtin.import_role:
+    name: "devscripts"
+    tasks_from: "build_config.yml"
+
+- name: Check if devscript cluster exists
   ansible.builtin.import_role:
-    name: devscripts
+    name: "devscripts"
+    tasks_from: "110_check_ocp.yml"
+
+- name: Push OCP related DNS entries
+  become: true
+  notify: "Restart dnsmasq"
+  vars:
+    _cluster: "{{ cifmw_devscripts_config['cluster_name'] }}"
+    _pub_net_data: >-
+      {{
+        _libvirt_manager_networking.networks[cifmw_libvirt_manager_pub_net]
+      }}
+    _api_addr: >-
+      {% if _pub_net_data.network_v4 is defined -%}
+      {{
+        _pub_net_data.network_v4 | ansible.utils.nthhost(2)
+      }}
+      {% endif -%}
+      {% if _pub_net_data.network_v6 is defined -%}
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(2)
+      }}
+      {% endif -%}
+    _apps_addr: >-
+      {% if _pub_net_data.network_v4 is defined -%}
+      {{
+        _pub_net_data.network_v4 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+      {% if _pub_net_data.network_v6 is defined -%}
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+  ansible.builtin.copy:
+    dest: "/etc/cifmw-dnsmasq.d/ocp_dns.conf"
+    mode: "0644"
+    content: |
+      local=/{{ _cluster }}.openstack.lab/
+      domain={{ _cluster }}.openstack.lab
+      address=/api.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/.apps.{{ _cluster }}.openstack.lab/{{ _apps_addr }}
+      localise-queries
+    validate: "/usr/sbin/dnsmasq -C %s --test"
+
+- name: Configure NetworkManager
+  become: true
+  notify: Restart NetworkManager
+  ansible.builtin.copy:
+    dest: >-
+      /etc/NetworkManager/conf.d/{{ cifmw_reproducer_networkmanager_config }}
+    mode: "0644"
+    content: |
+      [main]
+      rc-manager=none
+      dns=none
+    validate: "NetworkManager --print-config %s"
+
+- name: Ensure services are reloaded
+  ansible.builtin.meta: flush_handlers
+
+- name: Ensure we use dnsmasq as resolver
```

Comment:
This just does not work. Unless you want to start messing with creating dummy interfaces the interface will remain DOWN until you start hooking things up to the bridge.

```
9: cifmw-osp_trunk: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 52:54:00:68:ae:92 brd ff:ff:ff:ff:ff:ff
    inet6 fd00:aaaa::1/64 scope global tentative 
       valid_lft forever preferred_lft forever
```

We need to set dns=dnsmasq above in NetworkManager and configure conditional forwarding for our domain.

Reply:
If we add both IPv4 and IPv6 - or condition it on wheater it is defined or not we fix the issue below as well.
server=/{{ _cluster }}.openstack.lab/{{ _libvirt_manager_networking.networks.ctlplane.gw_v4 }}
server=/{{ _cluster }}.openstack.lab/{{ _libvirt_manager_networking.networks.ctlplane.gw_v6 }}

Reply:
I'm modifying the previous PR to configure dnsmasq in NM with forwarders. I'll update that one to consume it properly.